lib/containers/container_images.pm

```diff
@@ -100,10 +108,13 @@ sub test_opensuse_based_image {
     die 'Argument $image not provided!'   unless $image;
     die 'Argument $runtime not provided!' unless $runtime;
 
+    my ($host_version, $host_sp, $host_id) = get_os_release();
+    my ($image_version, $image_sp, $image_id) = get_os_release("$runtime run --rm $image");
+
     $version = 'Tumbleweed' if ($version =~ /^Staging:/);
 
-    # It is the right version
-    if ($distri eq 'sle') {
+    # SUSEConnect zypper service is supported only on SLE based image on SLE host
+    if ($host_id =~ 'sles' && $image_id =~ 'sles') {
```

Comment:
The check for `PRETTY_NAME` is still useful/needed though, so this should only check for `$image_id =~ 'sles'` here. The `$host_id =~ 'sles'` check is then only needed around the use of suseconnect.

This also fixes that testing a sles image on a !sles host doesn't end up in the openSUSE code below in the `else` branch.

Reply:
Now I see it. It should be fixed now so the SLE check is performed when the image is SLE but the zypper service part has another nested condition checking that also host is SLE.